opentelemetry-api/src/opentelemetry/context/propagation/binaryformat.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+
+from opentelemetry.trace import SpanContext
+
+
+class BinaryFormat(abc.ABC):
```

Comment:
I think this class could use some "performance-oriented" APIs: Since creating a slice of `bytes` copies it, I suggest `from_bytes(byte_representation:bytes, offset: int = 0, length: int = -1)`, and maybe `append_bytes(dst: bytearray)`.

Reply:
I can see the value here, but I may wait on adding these until I add an implementation (e.g. the standard binary format).  That might better illustrate the right API, unless you already have an example in mind.

Alternatively it looks like memoryview would be a great way to read values:

https://docs.python.org/3/library/stdtypes.html#memoryview

Although I could always construct a memoryview from the bytes object I received. For to_bytes I could use bytearray internally to construct the buffer.

Reply:
In case it's helpful, OC uses memoryview in the tags (read: distributed context) propagator: https://github.com/census-instrumentation/opencensus-python/blob/577fb61ab26513e3b448b6696432eefb0b11fa73/opencensus/tags/propagation/binary_serializer.py#L38.